modules/Microsoft.Purview/accounts/deploy.bicep

```diff
@@ -0,0 +1,394 @@
+param storageAccountQueuePrivateEndpointName string = ''
+
+@description('Optional. The custom name of the network interface attached to the managed Storage Account queue private endpoint.')
+param storageAccountQueuePrivateEndpointNicName string = ''
+
+@description('Optional. The static private IP address for the managed Storage Account blob private endpoint.')
+param storageAccountQueuePrivateEndpointIP string = ''
+
+@description('Conditional. Name of the managed Event Hub Namespace Private Endpoint. Required if the managed Event Hub Namespace private endpoint is required.')
+param eventHubPrivateEndpointName string = ''
+
+@description('Optional. The custom name of the network interface attached to the managed Event Hub Namespace private endpoint.')
+param eventHubPrivateEndpointNicName string = ''
+
+@description('Optional. The static private IP address for the managed Event Hub Namespace private endpoint.')
+param eventHubPrivateEndpointIP string = ''
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
+
+@description('Optional. The name of logs that will be streamed. "allLogs" includes all possible logs for the resource.')
+@allowed([
+  'allLogs'
+  'ScanStatus'
+  'DataSensitivity'
+  'PurviewAccountAuditEvents'
+])
+param diagnosticLogCategoriesToEnable array = [
+  'allLogs'
+]
+
+@description('Optional. The name of metrics that will be streamed.')
+@allowed([
+  'AllMetrics'
+])
+param diagnosticMetricsToEnable array = [
+  'AllMetrics'
+]
+
+@description('Optional. The name of the diagnostic setting, if deployed.')
+param diagnosticSettingsName string = '${name}-diagnosticSettings'
+
+@allowed([
+  ''
+  'CanNotDelete'
+  'ReadOnly'
+])
+@description('Optional. Specify the type of lock.')
+param lock string = ''
+
+// =========== //
+// Variables   //
+// =========== //
+var diagnosticsLogsSpecified = [for category in filter(diagnosticLogCategoriesToEnable, item => item != 'allLogs'): {
+  category: category
+  enabled: true
+  retentionPolicy: {
+    enabled: true
+    days: diagnosticLogsRetentionInDays
+  }
+}]
+
+var diagnosticsLogs = contains(diagnosticLogCategoriesToEnable, 'allLogs') ? [
+  {
+    categoryGroup: 'allLogs'
+    enabled: true
+    retentionPolicy: {
+      enabled: true
+      days: diagnosticLogsRetentionInDays
+    }
+  }
+] : diagnosticsLogsSpecified
+
+var diagnosticsMetrics = [for metric in diagnosticMetricsToEnable: {
+  category: metric
+  timeGrain: null
+  enabled: true
+  retentionPolicy: {
+    enabled: true
+    days: diagnosticLogsRetentionInDays
+  }
+}]
+
+var identityType = !empty(userAssignedIdentities) ? 'SystemAssigned,UserAssigned' : 'SystemAssigned'
+
+var identity = identityType != 'None' ? {
+  type: identityType
+  userAssignedIdentities: !empty(userAssignedIdentities) ? userAssignedIdentities : null
+} : null
+
+var deploymentNameSuffix = last(split(deployment().name, '-'))
```

Comment:
Not against changes, but would suggest another issue that describe CARML design pattern changes. Please remove this variable to align PE implementation names to CARML: 
```bicep
'${uniqueString(deployment().name, location)}-AutomationAccount-PrivateEndpoint-${index}'
```

Reply:
updated